src/app/core/panels/my-apps-panel/my-apps-panel.component.ts

```diff
@@ -401,57 +402,100 @@ export class MyAppsPanelComponent extends DirectiveBase implements OnInit, After
+                        click: () => {
+                            const modal = this.modal.fromComponent(CreateAppModalComponent, {
+                                closeOnOutsideClick: false,
+                                backdrop: true,
+                                title: `Create a New App`,
+                                closeOnEscape: true
+                            });
+
+                            modal.appType = 'tool';
+                            modal.destination = 'remote';
+                            modal.defaultProject = data.node.id;
+                        }
+                    })
+                ];
                 this.context.showAt(data.node.getViewContainer(), contextMenu, data.coordinates);
             });
 
         // When click on some root local folder
         this.tree.contextMenu.filter((data) => data.node.type === "folder" && data.node.level === 2)
             .subscribe(data => {
-                const contextMenu = [new MenuItem("Remove from Workspace", {
-                    click: () => {
-                        this.preferences.get("localFolders", []).take(1).subscribe(openFolders => {
-
-                            this.preferences.put("localFolders", openFolders.filter((el) => el !== data.node.id));
-                        });
-                    }
-                })].concat(contextMenuStaticOptions);
+                const contextMenu = [
+                    new MenuItem("Create new Folder", {
+                        click: () => {
+                            const modal = this.modal.fromComponent(CreateLocalFolderModalComponent, {
+                                closeOnOutsideClick: false,
+                                backdrop: true,
+                                title: `Create New Folder`,
+                                closeOnEscape: true
+                            });
+
+                            modal.folderPath = data.node.id;
+                        }
+                    }),
+                    new MenuItem("Remove from Workspace", {
+                        click: () => {
+                            this.preferences.get("localFolders", []).take(1).subscribe(openFolders => {
+
+                                this.preferences.put("localFolders", openFolders.filter((el) => el !== data.node.id));
+                            });
+                        }
+                    }),
+                    new MenuItem("Create new Workflow", {
+                        click: () => {
+                            const modal = this.modal.fromComponent(CreateAppModalComponent, {
+                                closeOnOutsideClick: false,
+                                backdrop: true,
+                                title: `Create a New App`,
+                                closeOnEscape: true
+                            });
+
+                            modal.appType = 'workflow';
+                            modal.defaultFolder = data.node.id;
```

Comment:
Setting the `defaultFolder` doesn't seem to do anything. When I create a new app, the folder which is selected regardless of how I open the "Create App Modal" is my home directory.